ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywallView.kt

```diff
@@ -55,6 +55,9 @@ private fun getPaywallViewModel(
 ): PaywallViewModel {
     val applicationContext = LocalContext.current.applicationContext
     return viewModel<PaywallViewModelImpl>(
+        // We need to pass the key in order to create different view models for different offerings when
+        // trying to load different paywalls for the same view model store owner.
+        key = offering?.identifier,
```

Comment:
This has an important drawback which is that, if a user displays multiple different offerings as dialogs, the viewmodels for all of them will remain in memory... I'm not sure if this is the best way to fix this, but it should work for now.

Reply:
I think that makes sense. It's also probably extremely uncommon.